### PR TITLE
Rename functions, removing gutenberg_ prefix

### PR DESCRIPTION
**Pull request by mkaz**
## Description

Renames two functions in latest-comments block PHP code.
Removes `gutenberg_` prefix

See #12085 

## How has this been tested?

* Confirmed latest comments block loads
* Confirmed no other function name conflicts

## Types of changes

* Renames two internal functions to latest comments block
* Updates reference to the function in lib/load.php

## Checklist:
- [X] My code is tested.
- [X] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

**Comment by swissspidy** [comment]
Does the Latest Posts block also use the wp_ prefix in core? 🤔 I don‘t think so.

We should aim for consistency here and fix this also in core before 5.0 is out.

**Review by swissspidy — CHANGES_REQUESTED** [review]
Other functions in WordPress are currently named `render_block_core_latest_posts()` without the `wp_` prefix.

**Comment by mkaz** [comment]
@swissspidy I changed them back without the prefix.
I was trying to address the concerns in #12085 but fine to match 👍

File: packages/block-library/src/latest-comments/index.php
Comment (swissspidy) [inline packages/block-library/src/latest-comments/index.php]: We should remove this check. We don't do pluggable functions in core anymore. Plus, it's the only `functions_exists()` check in this file.

File: packages/block-library/src/latest-comments/index.php
Comment (swissspidy) [inline packages/block-library/src/latest-comments/index.php]: This is a handy little utility function we could use in lots of places in core.

Any chances we could make this more generic, e.g. just `wp_draft_or_post_title()`? We could add this to WordPress core and remove it from Gutenberg even.

cc @pento

Comment (mkaz) [reply packages/block-library/src/latest-comments/index.php]: @swissspidy according to the comment, that function should already be in core, but now looking for it I can't find it in `wp-includes/template.php` or other spots.

File: packages/block-library/src/latest-comments/index.php
Comment (mkaz) [reply packages/block-library/src/latest-comments/index.php]: 👍

Comment (youknowriad) [reply packages/block-library/src/latest-comments/index.php]: Will this create issues when we run Gutenberg (the plugin) in Core? Loading the same function twice? We'll have to check.

File: packages/block-library/src/latest-comments/index.php
Comment (swissspidy) [reply packages/block-library/src/latest-comments/index.php]: These functions are all in `src/wp-includes/blocks/latest-comments.php`.

Comment (swissspidy) [reply packages/block-library/src/latest-comments/index.php]: That being said, `wp_latest_comments_draft_or_post_title()` could totally be renamed and moved to another, more generic file.

**Comment by youknowriad** [comment]
This tests well. @swissspidy what do you think we should do here to land it?

**Comment by swissspidy** [comment]
It's causing some pain because we already have the changes in core in some way. Including a ridiculous `functions_exists( 'wp_latest_comments_draft_or_post_title' )` check.

I think we can merge this as-is and then make sure core is updated accordingly.

**Comment by youknowriad** [comment]
The packages update will automatically get these changes to Core and sync the changes, this should be enough right? Just making sure I understand properly the potential issues here before merging.

**Comment by swissspidy** [comment]
Happy to get a second pair of eyes from @pento just to make sure, but yeah this should be enough.

**Comment by youknowriad** [comment]
@pento I'm merging this at the moment. If you have concerns, please let me know and I'll make the necessary changes.

**Review by youknowriad — APPROVED** [review]

**Comment by aduth** [comment]
The code which shipped with 5.0 defines these functions with a `gutenberg_` prefix. This is in error, but in the meantime the current Gutenberg master will cause a warning to be added to each page:

>Notice: WP_Block_Type_Registry::register was called incorrectly. Block type "core/latest-comments" is already registered. Please see Debugging in WordPress for more information. (This message was added in version 5.0.0.) in /srv/www/editor/htdocs/wp-includes/functions.php on line 4169

That's because this function is not in-fact defined with 5.0:

https://github.com/WordPress/gutenberg/blob/1a1dc7c15f556a5298c2549e219fdc79fb951304/lib/load.php#L65

**Comment by swissspidy** [comment]
We can certainly add a `function_exists()` check for the other function name in the meantime.

---

Related core ticket: https://core.trac.wordpress.org/ticket/45558

**Pull request merged**
